### DISCUSSION TITLE
How to use getList properly

### DISCUSSION THREAD

**Discussion by mbnoimi** [pocketbase/pocketbase#1773] [Q&A] [answered]
Using my Vue app I want to get first two items from the collection so I call the following. But it shows and empty entries!

BTW, I tried to use `pb?.collection('items').getFullList(2, {sort: '+created',});` it works fine but it shows all the entries.

```vue
<script setup lang="ts">
....
const items = ref([]);

async function getPrices() {
  const records = await pb?.collection('items').getList(1, 2, {
    sort: '+created',
  });
  if (records) {
    items.value = records;
  }
}
....
</script>
<template>
.....
        <div v-for="item in items" :key="item.id">
          &nbsp;&nbsp;&nbsp;{{ item.name }} = {{ item.price }}
        </div>
.....
</template>

```

**Comment by ganigeorgiev** [comment, marked as answer]
The resolved `Promise` of `getList` has the following format:

```js
{
    page:       number
    perPage:    number
    totalItems: number
    totalPages: number
    items:      Array<M>
}
```

Or in other words, you should be able to access the records doing something like:

```js
const result = await pb.collection('items').getList(1, 2, {
    sort: '+created',
});

items.value = result.items;
```

**Reply by ganigeorgiev** [reply]
_Also keep in mind that `result.items` in the above could be empty array if the request client doesn't satisfy the "List/Search API" rule of your collection._

**Reply by mbnoimi** [reply]
Oops, the auto completion doesn't show `result.items` thanks any way.
